fjord/l10n/utils.py

```diff
@@ -0,0 +1,9 @@
+import re
+
+
+_COLLAPSE_RE = re.compile(r'\s+', re.UNICODE)
+
+
+def collapse_whitespace(message):
+    """Collapses consecutive whitespace into a single space"""
+    return _COLLAPSE_RE.sub(' ', message).strip()
```

Comment:
That's pretty interesting. Django uses a regexp:

https://github.com/django/django/blob/master/django/utils/translation/__init__.py#L243

I like that your version is faster for the important/likely scenarios. My one gripe is that it's less readable to me, but that's pretty minor. I want to test it with Python 3, but then I'd be game for changing it.